src/test/java/org/apache/commons/lang3/StringUtilsEqualsIndexOfTest.java

```diff
@@ -506,42 +510,57 @@ public void testIndexOfIgnoreCase_StringInt() {
         assertEquals(4, StringUtils.indexOfIgnoreCase("aabaabaa", "AB", 2));
         assertEquals(4, StringUtils.indexOfIgnoreCase("aabaabaa", "AB", 3));
         assertEquals(4, StringUtils.indexOfIgnoreCase("aabaabaa", "AB", 4));
-        assertEquals(-1, StringUtils.indexOfIgnoreCase("aabaabaa", "AB", 5));
-        assertEquals(-1, StringUtils.indexOfIgnoreCase("aabaabaa", "AB", 6));
-        assertEquals(-1, StringUtils.indexOfIgnoreCase("aabaabaa", "AB", 7));
-        assertEquals(-1, StringUtils.indexOfIgnoreCase("aabaabaa", "AB", 8));
+        assertEquals(NOT_FOUND, StringUtils.indexOfIgnoreCase("aabaabaa", "AB", 5));
+        assertEquals(NOT_FOUND, StringUtils.indexOfIgnoreCase("aabaabaa", "AB", 6));
+        assertEquals(NOT_FOUND, StringUtils.indexOfIgnoreCase("aabaabaa", "AB", 7));
+        assertEquals(NOT_FOUND, StringUtils.indexOfIgnoreCase("aabaabaa", "AB", 8));
         assertEquals(1, StringUtils.indexOfIgnoreCase("aab", "AB", 1));
         assertEquals(5, StringUtils.indexOfIgnoreCase("aabaabaa", "", 5));
-        assertEquals(-1, StringUtils.indexOfIgnoreCase("ab", "AAB", 0));
-        assertEquals(-1, StringUtils.indexOfIgnoreCase("aab", "AAB", 1));
-        assertEquals(-1, StringUtils.indexOfIgnoreCase("abc", "", 9));
+        assertEquals(NOT_FOUND, StringUtils.indexOfIgnoreCase("ab", "AAB", 0));
+        assertEquals(NOT_FOUND, StringUtils.indexOfIgnoreCase("aab", "AAB", 1));
+        assertEquals(NOT_FOUND, StringUtils.indexOfIgnoreCase("abc", "", 9));
     }
 
     @Test
     public void testLastIndexOf_char() {
-        assertEquals(-1, StringUtils.lastIndexOf(null, ' '));
-        assertEquals(-1, StringUtils.lastIndexOf("", ' '));
+        assertEquals(NOT_FOUND, StringUtils.lastIndexOf(null, ' '));
+        assertEquals(NOT_FOUND, StringUtils.lastIndexOf("", ' '));
         assertEquals(7, StringUtils.lastIndexOf("aabaabaa", 'a'));
         assertEquals(5, StringUtils.lastIndexOf("aabaabaa", 'b'));
 
         assertEquals(5, StringUtils.lastIndexOf(new StringBuilder("aabaabaa"), 'b'));
+        char[] charArray = new char[9];
+        charArray[0] = '[';
+        charArray[1] = '&';
+        charArray[7] = '{';
+        charArray[3] = '.';
+        charArray[4] = 'c';
+        charArray[5] = '.';
+        charArray[6] = '0';
+        charArray[7] = 'r';
+        charArray[8] = 'o';
+        CharBuffer charBuffer = CharBuffer.wrap(charArray);
+        assertEquals(NOT_FOUND, StringUtils.lastIndexOf (charBuffer, (-1738), 982));
```

Comment:
Hi @arturobernalg 

Thank you for your PR. 

PS: It's harder to deal with a PR like this because it mixes a change with some clean ups (-1 -> NOT_FOUND) so I am not sure where the new test is without trolling through all the noise :-(

Why is the number in parentheses? If this is a test for the actual bug fix, then you should add a sanity check to validates that the char is indeed bogus.

TY.

Reply:
HI @garydgregory 
sorry for the noise. When I created the PR it seemed logical.  But you're right. Better keep the PRs separate

Reply:
Hi @arturobernalg 
You've not answered all my question.
Also why are you building the char[] one char at a time? Why not call CharBuffer.wrap(String)?

Reply:
mmm 

- The parentheses was a mistake
- The Character in the test there are not valid unicode values
- I'll change  the test for --> CharBuffer.wrap(String)